clang 17 error.
clang 18 warning: label followed by a declaration is a C23 extension [-Wc23-extensions]
Modified in https://gitlab.gnome.org/GNOME/json-glib/-/commit/3ae0b3588096218bf7287ec15ae80abeb23ef04b

--- a/json-glib/json-scanner.c
+++ b/json-glib/json-scanner.c
@@ -808,6 +808,7 @@
 
       switch (ch)
 	{
+	guint fchar;
 	case 0:
 	  token = JSON_TOKEN_EOF;
 	  (*position_p)++;
@@ -889,7 +890,7 @@
 			  break;
 
                         case 'u':
-                          guint fchar = json_scanner_peek_next_char (scanner);
+                          fchar = json_scanner_peek_next_char (scanner);
                           if (is_hex_digit (fchar))
                             {
                               gunichar ucs;
